model/src/main/java/com/javahelp/model/user/UserPassword.java

```diff
@@ -1,63 +1,62 @@
-package com.javahelp.model.userEntityTest;
+package com.javahelp.model.user;
 
 import java.util.Base64;
 
+/**
+ * A class for the password of a user entity
+ */
 public class UserPassword {
+
     /**
-     * A class for hashing a user's password
      * --- Attributes ---
      * salt - a current salt of a user
      * hash - the current and hashed password of the user
-     * user - the user object
      */
-
     private byte[] salt;
 
     private byte[] hash;
 
-    private User user;
-
     /** The constructor to build an instance of the password of a user account.
      *
      * @param salt - The currently used salt for the hashed password.
      * @param hash - The already-hashed password
-     * @param user - the current user account
      */
-
-    public UserPassword(User user, byte[] salt, byte[] hash){
-        this.user = user;
+    public UserPassword(byte[] salt, byte[] hash){
         this.salt = salt;
         this.hash = hash;
     }
 
-    /** Getters of the current attributes of UserPassword
+    /** Get the current hashed password of UserPassword
      *
-     * @return Hash - the currently hashed password
-     *         Salt - the currently used salt
-     *         User - the current user
+     * @return The currently hashed password
      */
-
     public byte[] getHash() {
         return hash;
     }
 
+    /** Get the current salt of UserPassword
+     *
+     * @return The currently used salt
+     */
     public byte[] getSalt() {
         return salt;
     }
 
-
-    /** The setters of the current attributes. Please note that only current salt and
-     * password can be set, while the current user always stays the same
-     * in this UserPassword entity!
+    /** Set current hashed password to the user.
+     * Please note that only hashed password can be set, while the current user always
+     * stays the same in this UserPassword entity!
```

Comment:
This is true, but really the `UserPassword` just doesn't have any reference to a `User` so it's more that `UserPassword` is user-agnostic than that the user is always the same